samples/extensions/dynamic_multisample_rasterization/dynamic_multisample_rasterization.cpp

```diff
@@ -0,0 +1,947 @@
+	auto &requested_dynamic_rendering            = gpu.request_extension_features<VkPhysicalDeviceDynamicRenderingFeaturesKHR>(VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DYNAMIC_RENDERING_FEATURES_KHR);
+	requested_dynamic_rendering.dynamicRendering = VK_TRUE;
+}
+
+const std::string to_string(VkSampleCountFlagBits count)
+{
+	switch (count)
+	{
+		case VK_SAMPLE_COUNT_1_BIT:
+			return "No MSAA";
+		case VK_SAMPLE_COUNT_2_BIT:
+			return "2X MSAA";
+		case VK_SAMPLE_COUNT_4_BIT:
+			return "4X MSAA";
+		case VK_SAMPLE_COUNT_8_BIT:
+			return "8X MSAA";
+		case VK_SAMPLE_COUNT_16_BIT:
+			return "16X MSAA";
+		case VK_SAMPLE_COUNT_32_BIT:
+			return "32X MSAA";
+		case VK_SAMPLE_COUNT_64_BIT:
+			return "64X MSAA";
+		default:
+			return "Unknown";
+	}
+}
+
+void DynamicMultisampleRasterization::prepare_supported_sample_count_list()
+{
+	if (sample_count_prepared)
+		return;
+
+	VkPhysicalDeviceProperties gpu_properties;
+	vkGetPhysicalDeviceProperties(get_device().get_gpu().get_handle(), &gpu_properties);
+
+	VkSampleCountFlags supported_by_depth_and_color = gpu_properties.limits.framebufferColorSampleCounts & gpu_properties.limits.framebufferDepthSampleCounts;
+
+	// All possible sample counts are listed here from most to least preferred as default
+	// On Mali GPUs 4X MSAA is recommended as best performance/quality trade-off
+	std::vector<VkSampleCountFlagBits> counts = {VK_SAMPLE_COUNT_4_BIT, VK_SAMPLE_COUNT_2_BIT, VK_SAMPLE_COUNT_8_BIT,
+	                                             VK_SAMPLE_COUNT_16_BIT, VK_SAMPLE_COUNT_32_BIT, VK_SAMPLE_COUNT_64_BIT,
+	                                             VK_SAMPLE_COUNT_1_BIT};
+
+	std::copy_if(counts.begin(),
+	             counts.end(),
+	             std::back_inserter(supported_sample_count_list),
+	             [&supported_by_depth_and_color](auto count) { return supported_by_depth_and_color & count; });
+	std::transform(supported_sample_count_list.begin(),
+	               supported_sample_count_list.end(),
+	               std::back_inserter(gui_settings.sample_counts),
+	               [](auto count) { return to_string(count); });
+	if (!supported_sample_count_list.empty())
+	{
+		sample_count = supported_sample_count_list.front();
+	}
+
+	sample_count_prepared = true;
+}
+
+bool DynamicMultisampleRasterization::prepare(const vkb::ApplicationOptions &options)
+{
+	if (!ApiVulkanSample::prepare(options))
+	{
+		return false;
+	}
+
+#if VK_NO_PROTOTYPES
+	VkInstance instance = get_device().get_gpu().get_instance().get_handle();
+	assert(!!instance);
+	vkCmdBeginRenderingKHR = reinterpret_cast<PFN_vkCmdBeginRenderingKHR>(vkGetInstanceProcAddr(instance, "vkCmdBeginRenderingKHR"));
+	vkCmdEndRenderingKHR   = reinterpret_cast<PFN_vkCmdEndRenderingKHR>(vkGetInstanceProcAddr(instance, "vkCmdEndRenderingKHR"));
+	if (!vkCmdBeginRenderingKHR || !vkCmdEndRenderingKHR)
+	{
+		throw std::runtime_error("Unable to dynamically load vkCmdBeginRenderingKHR and vkCmdEndRenderingKHR");
+	}
+#endif
```

Comment:
Why do you get those function pointers here? They are already available

Reply:
Removed.